IPython/terminal/pt_inputhooks/__init__.py

```diff
@@ -40,6 +39,80 @@ def __str__(self):
                                     ', '.join(backends + sorted(registered)))
 
 
+def set_qt_api(gui):
+    """Sets the `QT_API` environment variable if it isn't already set."""
+
+    qt_api = os.environ.get("QT_API", None)
+
+    from IPython.external.qt_loaders import (
+        QT_API_PYQT,
+        QT_API_PYQT5,
+        QT_API_PYQT6,
+        QT_API_PYSIDE,
+        QT_API_PYSIDE2,
+        QT_API_PYSIDE6,
+        QT_API_PYQTv1,
+        loaded_api,
+    )
+
+    loaded = loaded_api()
+
+    qt_env2gui = {
+        QT_API_PYSIDE: "qt4",
+        QT_API_PYQTv1: "qt4",
+        QT_API_PYQT: "qt4",
+        QT_API_PYSIDE2: "qt5",
+        QT_API_PYQT5: "qt5",
+        QT_API_PYSIDE6: "qt6",
+        QT_API_PYQT6: "qt6",
+    }
+    if loaded is not None and gui != "qt":
+        if qt_env2gui[loaded] != gui:
+            raise ImportError(
+                f"Cannot switch Qt versions for this session; must use {qt_env2gui[loaded]}."
+            )
```

Comment:
Why not change this `raise` by a `print` instead? I think it's not a good UX to show a long traceback for this, i.e. users don't need to receive that kind of info, only that IPython can't switch to another Qt version or binding.